### PR TITLE
Added action recognition model

### PR DESCRIPTION
**Pull request by vadimadr**
**Before submitting a pull request**, please make sure the following is done:

1. You have merged in the latest changes from `develop`.
2. Resolved any conflicts that may have happened from that merge.
3. Ensure that you have run all tests and linting locally.

### CODE REVIEW

**Comment by vadimadr** [comment]
cc @snosov1 @AlexanderDokuchaev

File: pytorch_toolkit/action_recognition/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+4. [Models](#models)
+5. [Action Recognition Demo](#demo)
+
+
+## Requirements
+
+The code is tested on Python 3, with dependencies listed in `requirements.txt` file. You can install required packages with:
+
+```bash
+pip intall -r requirements.txt
```
Comment (DmitriySidnev) [inline pytorch_toolkit/action_recognition/README.md]: Missed letter `s`: pip in**s**tall -r requirements.txt

File: pytorch_toolkit/action_recognition/utils/preprocess_videos.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    if annotation['subset'] == 'testing':
+        label = 'test'
+    else:
+        label = annotation['annotations']['label']
+
+    class_path = os.path.join(destination_path, label)
+    Path(class_path).mkdir(exist_ok=True)
+
+    orig_video_path = os.path.join(orig_path, label, video_name) + ".mp4"
```
Comment (AlexanderDokuchaev) [inline pytorch_toolkit/action_recognition/utils/preprocess_videos.py]: It's not work for  `UCF-101` dataset that contains video with `.avi` extension

File: pytorch_toolkit/action_recognition/action_recognition/utils.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+        'optimizer': optimizer.state_dict(),
+    }
+    torch.save(states, save_file_path.as_posix())
+
+
+def create_code_snapshot(root, dst_path, extensions=(".py", ".json")):
+    """Creates tarball with the source code"""
+    with tarfile.open(str(dst_path), "w:gz") as tar:
+        for path in Path(root).rglob("*"):
+            if '.git' in path.parts:
```
[the remark this answers is not in the document]
Comment (vadimadr) [reply pytorch_toolkit/action_recognition/action_recognition/utils.py]: I'd say it is a feature, but not intended. It might be useful to track versions of the packages for reproducibility, so i'll keep oit

File: pytorch_toolkit/action_recognition/devtools/pylint.rc [hunk 4]
```diff
@@ -0,0 +1,5 @@
+[MASTER]
+disable = all
```
[the remark this answers is not in the document]
Comment (vadimadr) [reply pytorch_toolkit/action_recognition/devtools/pylint.rc]: I replaced it with an actual list of exception. Maybe they'll be resolved some day, but I won't do it right now.

File: pytorch_toolkit/action_recognition/utils/preprocess_videos.py (hunk 2 above)
Comment (vadimadr) [reply pytorch_toolkit/action_recognition/utils/preprocess_videos.py]: thanks, fixed

**Comment by vadimadr** [comment]
@AlexanderDokuchaev resolved discussions.

File: pytorch_toolkit/action_recognition/README.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+This is the implementation of **Video Transformer Network** approach for Action Recognition in PyTorch. The repository also contains training code for other action recognition models, such as 3D CNNs, LSTMs, I3D, R(2+1)D, Two stream networks. 
+
+## Table of Contents
+
+1. [Requirements](#requirements)
+2. [Preparation](#preparation)
+3. [Train/Eval](#traineval)
+4. [Models](#models)
+5. [Action Recognition Demo](#demo)
```
[the remark this answers is not in the document]
Comment (vadimadr) [reply pytorch_toolkit/action_recognition/README.md]: But there is demo section in README...
